### PR TITLE
Allow Pypp to work with arrays of DataVectors

### PR DESCRIPTION
**Pull request by osheamonn**
## Proposed changes

The characteristic speeds in Cris' PR will be held as a `std::array` so we need to be able to test arrays with the python infrastructure. I also had to add a `make_with_value` implementation for arrays to get this to work.  

### Types of changes:

- [ ] Bugfix
- [ ] New feature

### Component:

- [ ] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- Follows [code review guidelines](https://sxs-collaboration.github.io/spectre/code_review_guide.html)
- Code has documentation and unit tests
- Private member variables have a trailing underscore
- Do not use [Hungarian notation](https://en.wikipedia.org/wiki/Hungarian_notation), e.g. `double* pd_blah` is bad
- Header order:
  1. hpp corresponding to cpp (only in cpp files)
  2. Blank line (only in cpp files)
  3. STL and externals (in alphabetical order)
  4. Blank line
  5. SpECTRE includes (in alphabetical order)
- File lists in CMake are alphabetical
- Correct `noexcept` specification for functions (if unsure, mark `noexcept`)
- Mark objects `const` whenever possible
- Almost always `auto`, except with expression templates, i.e. `DataVector`
- All commits for performance changes provide quantitative evidence and the tests used to obtain said evidence.
- Make sure error messages are helpful, e.g. "The number of grid points in the matrix 'F' is not the same as the number of grid points in the determinant."
- Prefix commits addressing PR requests with `fixup`. These commits are flagged
  by Travis so we do not accidentally merge them.
- Include what you use, prefer forward declarations in hpp files.
- Explicitly make numbers floating point, e.g. `2.` or `2.0` over `2`
- Use `Tensor`'s non-member get if the indices are constant expressions, e.g.
  `get<1>(tensor)`

### CODE REVIEW

File: src/Utilities/MakeWithValue.hpp [hunk 1]
```diff
@@ -6,8 +6,11 @@
 
 #pragma once
 
+#include <array>
+
 #include "Utilities/ForceInline.hpp"
 #include "Utilities/TaggedTuple.hpp"
+#include "Utilities/MakeArray.hpp"
```
Comment (nilsdeppe) [inline src/Utilities/MakeWithValue.hpp]: alphabet

File: tests/Unit/Pypp/Pypp.hpp [hunk 2]
```diff
@@ -54,9 +54,27 @@ struct CallImpl {
...
+
+template <typename T>
+using convert_to_container_of_doubles_t =
+    typename convert_to_container_of_doubles<T>::type;
+
 template <typename R>
-struct CallImpl<R, Requires<tt::is_a_v<Tensor, R> and
-                            cpp17::is_same_v<typename R::type, DataVector>>> {
+struct CallImpl<
+    R, Requires<(tt::is_a_v<Tensor, R> or tt::is_std_array_v<R>)and cpp17::
```
Comment (nilsdeppe) [inline tests/Unit/Pypp/Pypp.hpp]: the formatting here is odd

**Review by nilsdeppe — CHANGES_REQUESTED** [review]

File: tests/Unit/Pypp/Pypp.hpp (hunk 2 above)
Comment (osheamonn) [reply tests/Unit/Pypp/Pypp.hpp]: Clang-format being weird... Will fix.

**Review by nilsdeppe — CHANGES_REQUESTED** [review]
Move alphabet correction to right commit and squash

**Review by nilsdeppe — APPROVED** [review]
@kidder

**Review by kidder — APPROVED** [review]

**Pull request merged**